satellite/orders/service.go

```diff
@@ -707,6 +707,72 @@ func (service *Service) CreatePutRepairOrderLimits(ctx context.Context, bucketID
 	return limits, piecePrivateKey, nil
 }
 
+// CreateGracefulExitPutOrderLimit creates an order limit for graceful exit put transfers.
+func (service *Service) CreateGracefulExitPutOrderLimit(ctx context.Context, bucketID []byte, nodeID storj.NodeID, pieceNum int32, rootPieceID storj.PieceID, shareSize int32) (limit *pb.AddressedOrderLimit, _ storj.PiecePrivateKey, err error) {
+	defer mon.Task()(&ctx)(&err)
+
+	orderExpiration := time.Now().UTC().Add(service.orderExpiration)
+
+	piecePublicKey, piecePrivateKey, err := storj.NewPieceKey()
+	if err != nil {
+		return nil, storj.PiecePrivateKey{}, Error.Wrap(err)
+	}
+
+	serialNumber, err := service.createSerial(ctx)
+	if err != nil {
+		return nil, storj.PiecePrivateKey{}, Error.Wrap(err)
+	}
+
+	node, err := service.overlay.Get(ctx, nodeID)
+	if err != nil {
+		return nil, storj.PiecePrivateKey{}, Error.Wrap(err)
+	}
+
+	if node.Disqualified != nil {
+		return nil, storj.PiecePrivateKey{}, overlay.ErrNodeDisqualified.New("%v", nodeID)
```

Comment:
Does this mean disqualified Node should not be able to exit gracefully?

Reply:
This node is the new node, or "replacement" node as we called it in the doc.  So we just want to make sure it's a valid node before creating the `AddressedOrderLimit`.